rs/cli/src/auth.rs

```diff
@@ -22,21 +32,23 @@ pub struct Neuron {
 static RELEASE_AUTOMATION_DEFAULT_PRIVATE_KEY_PEM: &str = ".config/dfx/identity/release-automation/identity.pem"; // Relative to the home directory
 const RELEASE_AUTOMATION_NEURON_ID: u64 = 80;
 
+// As per fn str_to_key_id(s: &str) in ic-canisters/.../parallel_hardware_identity.rs,
+// the representation of key ID that the canister client wants is a sequence of
+// pairs of hex digits, case-insensitive.  The key ID as stored in the HSM is
+// a Vec<u8>.  We only store the little-endianest of the digits from that Vec<> in
+// our key_id variable.  The following function produces what the ic-canisters
+// code wants.
+pub fn hsm_key_id_to_string(s: u8) -> String {
+    format!("{:02x?}", s)
+}
+
 impl Neuron {
-    pub async fn new(
-        private_key_pem: Option<PathBuf>,
-        hsm_slot: Option<u64>,
-        hsm_pin: Option<String>,
-        hsm_key_id: Option<String>,
-        neuron_id: Option<u64>,
-        network: &Network,
-        include_proposer: bool,
-    ) -> anyhow::Result<Self> {
-        let auth = Auth::from_cli_args(private_key_pem, hsm_slot, hsm_pin, hsm_key_id)?;
+    pub async fn new(auth: Auth, neuron_id: Option<u64>, network: &Network, include_proposer: bool) -> anyhow::Result<Self> {
         let neuron_id = match neuron_id {
             Some(n) => n,
-            None => auth.auto_detect_neuron_id(network.get_nns_urls()).await?,
+            None => auth.auto_detect_neuron_id(network.get_nns_urls().to_vec()).await?,
         };
+        info!("Identifying as neuron ID {}", neuron_id);
```

Comment:
I think our tool is becoming way too verbose, I suggest change  this to debug/trace